Navigation, SVG Support, Home Brewed Styled Components

conrad-vanl opened the pull request:
This adds in all of the existing icons. 

Closes #52 

## Creator

- [x] Build relevant tests if any apply
- [x] Ensure there are no new warnings
- [ ] Upload an animated GIF showcasing the solution (if it applies)
- [x] Set a relevant reviewer

## Reviewer

- [x] Run `test` and make sure all tests pass
- [x] Review function and form on iOS
- [x] Review function and form on Android
- [x] Review function and form on Web
- [x] Review new test logic

mary-pr-poppins commented:
<!--
  0 failure: 
  3 warning:  Please assign som..., No Changelog chan..., :exclamation: Big PR
  
  
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please assign someone to merge this PR, and optionally include people who should review.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  No Changelog changes!
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  :exclamation: Big PR
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

burkeshartsis commented on src/@primitives/withTheme.js:
```diff
@@ -1,17 +1,20 @@
 import getContext from 'recompose/getContext';
 import compose from 'recompose/compose';
-import withProps from 'recompose/withProps';
+import mapProps from 'recompose/mapProps';
```
Is there a reason these aren't just imported together like this? 

``` 
import {
  getContext,
  compose,
  mapProps,
} from 'recompose';
```

burkeshartsis commented on src/@primitives/MediaQuery/enhancer.js:
```diff
@@ -0,0 +1,48 @@
+
+// Make it easy to do breakpoint-based component branching, a-la recompose's branch method.
+// The "test" function should return an object that mirrors what you'd do in a css mediaQuery:
+// import { enhancer as mediaQuery } from 'MediaQuery';
```
I like your example here of importing `enhancer as mediaQuery` but I'd rather see us just name this explicitly and export it that way so that people are never left importing something as vaguely named as `enhancer` or left to name it themselves and be inconsistent with it. Maybe name this `MediaQuery` and rename the media query component to something like `<Breakpoint...>`. Just a thought. Not tied to it.

conrad-vanl replied:
I thought about this for a while. If I think of `MediaQuery` as a module, that module exposes two pieces of API that you can consume - a component (default) and an enhancer.

The default export isn't named, cuz it's the default export. In the sense of the module, calling the only enhancer that it exposes "enhancer", was the most accurate and consistent thing I could think of doing. Almost like saying `MediaQuery.enhancer`.

Changing the component name to `Breakpoint` and calling the enhancer `mediaQuery` also felt odd...because doing that we'd have to change the name of the `MediaQuery` folder to `Breakpoint`, then it becomes the `Breakpoint` module exposing a `component` (which we call `Breakpoint`), and an enhancer (which we call `mediaQuery`). See where I'm going?

I think the standard API for doing `MediaQuery`s is the component. The alternative API for doing `MediaQuery`s would be MediaQuery's enhancer. 

Naming sucks.

Yeah...I'm open to changing this to something different, just not sure I like that suggestion either.

conrad-vanl commented on src/@primitives/Icon/index.js:
```diff
@@ -0,0 +1,30 @@
+});
+
+Icon.propTypes = {
+  name: PropTypes.oneOf(Object.keys(Icons).map(kebabCase)).isRequired,
```
going different directions. Icon names are supplied in kebabCase:

`<Icon name="my-icon" />`

Where the modules are currently exported in `PascalCase`

burkeshartsis commented on src/@primitives/Icon/index.js:
```diff
@@ -0,0 +1,30 @@
+import { compose, pure } from 'recompose';
+import { flow, camelCase, upperFirst, kebabCase } from 'lodash';
+
+import * as Icons from './icons';
```
@conrad-vanl Are there any performance issues with this? If I import `<Icon...>` I actually end up importing every icon.

Lepozepo replied:
This is a good place for dynamic imports :D but I’m not sure if we support them

Lepozepo replied:
@burkeshartsis it has a minor impact on web I think but I don’t think it’s the worst thing in the world

burkeshartsis replied:
I'm not super concerned I was just thinking about it tonight and was wondering if it was something we should be concerned about. Dynamic imports would be great if RN supported them 😢

conrad-vanl replied:
> Maybe we should split it into a .web.js and have that one do dynamic import?
Later optimization.

Lepozepo commented:
Maybe we should split it into a .web.js and have that one do dynamic import?

On Thu, Nov 9, 2017 at 6:01 PM Burke Shartsis <notifications@github.com>
wrote:

> *@burkeshartsis* commented on this pull request.
> ------------------------------
>
> In src/@primitives/Icon/index.js
> <https://github.com/NewSpring/Apollos/pull/87#discussion_r150138790>:
>
> > @@ -0,0 +1,30 @@
> +import PropTypes from 'prop-types';
> +import React from 'react';
> +import { compose, pure } from 'recompose';
> +import { flow, camelCase, upperFirst, kebabCase } from 'lodash';
> +
> +import * as Icons from './icons';
>
> I'm not super concerned I was just thinking about it tonight and was
> wondering if it was something we should be concerned about. Dynamic imports
> would be great if RN supported them 😢
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/NewSpring/Apollos/pull/87#discussion_r150138790>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ACMgnnGztAPZFkX98n7FcHPhJKiJpYRKks5s065bgaJpZM4QVLpP>
> .
>
-- 
--Marcelo Reyna--

conrad-vanl commented on src/@primitives/MediaQuery/withWindow.js:
```diff
@@ -0,0 +1,26 @@
+import React, { PureComponent } from 'react';
+import { Dimensions } from 'react-native';
+
+export default function withWindow(ComponentToWrap) {
```
I'm not sure how

conrad-vanl commented on src/@primitives/MediaQuery/index.js:
```diff
@@ -1,72 +1,56 @@
+      md: PropTypes.number.isRequired,
+      lg: PropTypes.number.isRequired,
+    }).isRequired,
+    window: PropTypes.shape({
```
done.

conrad-vanl commented:
Alright...I _think_ I addressed all of your concerns @burkeshartsis. I also added in a few more changes, so you might look at the commits made since your review, starting at 18554ac ("swap in new svgs")
